clarity-ext-scripts/clarity_ext_scripts/covid/ct_discriminator.py

```diff
@@ -0,0 +1,66 @@
+from clarity_ext.domain.validation import UsageError
+
+
+VALID_ASSAYS = ['Assay 1']
+
+BOUNDARY_DICT = {'Assay 1': (10, 50)}
+
+
+class CtDiscriminator(object):
+    """
+    A service to analyze if a sample is positive for Covid19
+    Given:
+    CT value of sample
+    Discriminator for assay (lower bound, upper bound)
+    This service does not update any artifact from context, it only reads from it
+    """
+    def __init__(self, context):
+        self.context = context
+        self.initial_validation()
+
+    def initial_validation(self):
+        if not self._has_assay_udf():
+            raise UsageError("The udf 'Assay' must be filled in before running this script")
+        if self.assay not in VALID_ASSAYS:
+            raise UsageError("The current assay value is not recognized: {}"
+                             .format(self.assay))
+
+    @property
+    def assay(self):
+        return self.context.current_step.udf_assay
+
+    def _has_assay_udf(self):
+        try:
+            dummy = self.assay
```

Comment:
There should be no need to assign a dummy variable to get the desired result here.

Reply:
I think some code quality tools might complain if it's only `self.assay`, so perhaps `_ = self.assay`? Of course `hasattr` would work generally, but not entirely sure in this case, as this is our own UdfMapping stuff, not sure it supports that, because they use the black magic `get_attribute` family of methods.